src/plugin/window/resource/Window.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#include <string>
+
+#include <GLFW/glfw3.h>
+#include <stdexcept>
+
+#include "WindowError.hpp"
+
+namespace ES::Plugin::Window::Resource {
+    /**
+     * The Window class is a singleton to handle the creation and destruction of the game's window.
+     */
+    class Window
+    {
+        private:
+            int _width;
+            int _height;
+            std::string _title;
+            GLFWwindow* _window;
+            GLFWmonitor* _monitor;
+            GLFWwindow* _share;
```

Comment:
Do we put members attributes at the start or at the end of the class definition ?

Reply:
we'll see with the coding style issue #13, right now it doesn't matter
(i personnaly prefer declaring them on top)